src/EPR.Payment.Facade/Validations/RegistrationFees/ProducerFeesRequestDtoValidator.cs

```diff
@@ -24,20 +24,7 @@ public ProducerFeesRequestDtoValidator()
 
             RuleFor(x => x.Regulator)
                 .NotEmpty().WithMessage(ValidationMessages.RegulatorRequired)
```

Comment:
the code can be improved by combining the validations for NumberofSubsidiaries and also using CascadeMode.Stop

RuleFor(x => x.ProducerType)
            .Cascade(CascadeMode.Stop)
            .Must(pt => string.IsNullOrWhiteSpace(pt) || validProducerTypes.Contains(pt.ToUpperInvariant()))
            .WithMessage($"{ValidationMessages.ProducerTypeInvalid} {string.Join(", ", validProducerTypes)}");

        RuleFor(x => x.NumberOfSubsidiaries)
            .Cascade(CascadeMode.Stop)
            .GreaterThanOrEqualTo(0).WithMessage(ValidationMessages.NumberOfSubsidiariesRange)
            .GreaterThan(0)
            .When(x => string.IsNullOrEmpty(x.ProducerType))
            .WithMessage(ValidationMessages.NumberOfSubsidiariesRequiredWhenProducerTypeEmpty);

        RuleFor(x => x.Regulator)
            .Cascade(CascadeMode.Stop)
            .NotEmpty().WithMessage(ValidationMessages.RegulatorRequired)
            .Must(RegulatorValidationHelper.IsValidRegulator).WithMessage(ValidationMessages.RegulatorInvalid);

Reply:
they want to see all messages like <Regulator is required.; Invalid Regulator.>

Reply:
any reason why they can't be combined? Also use cascade mode for the other validations

Reply:
First we need to check if NumberOfSubsidiaries is greater than or equal zero. Then we need to check that if ProducerType is null or empty then NumberOfSubsidiaries must be greater than zero. we should check them separatelly.  

There is no need to use cacade since each rule returns a message